### ISSUE TITLE
Registration failing due to missing xlAutoFree12 for "xlOil_Utils" and "xlOil_SQL" 

### ISSUE THREAD

**Issue by andyvandy** [cunnane/xloil#56]
( Firstly thanks for all the hardwork put into this project! I've recently been trialing xlOil and have been blown away. )

I've recently begun encountering errors at launch due to missing xlAutoFree12.

<details>
  <summary>Sample Logs</summary>

[2023-08-24 16:58:36.920] [logger] [info] [PluginLoader.cpp:78] Loading plugin xlOil_Utils
[2023-08-24 16:58:36.932] [logger] [debug] [register.py:110] Interpreted arg 'ModuleName:str' => 'ModuleName:str ()'
[2023-08-24 16:58:36.932] [logger] [debug] [register.py:110] Interpreted arg 'From:=None' => 'From:object ()'
[2023-08-24 16:58:36.932] [logger] [debug] [register.py:110] Interpreted arg 'As:=None' => 'As:object ()'
[2023-08-24 16:58:36.933] [logger] [debug] [register.py:431] Declared excel func: xloImport(ModuleName: str, From: object, As: object)
[2023-08-24 16:58:36.936] [logger] [debug] [Throw.cpp:17] Module 'xlOil_Utils' must define xlAutoFree12 to register Excel functions (in FuncRegistry.cpp:106 during xloil::FunctionRegistry::registerWithExcel)
[2023-08-24 16:58:36.939] [logger] [debug] [register.py:110] Interpreted arg 'Object:' => 'Object:object ()'
[2023-08-24 16:58:36.944] [logger] [debug] [FuncRegistry.cpp:299] Deregistering xloBlock
[2023-08-24 16:58:36.944] [logger] [warning] [FuncRegistry.cpp:304] Unregister failed for xloBlock
[2023-08-24 16:58:36.944] [logger] [debug] [register.py:110] Interpreted arg 'Name:str' => 'Name:str ()'
[2023-08-24 16:58:36.945] [logger] [debug] [Throw.cpp:17] Module 'xlOil_Utils' must define xlAutoFree12 to register Excel functions. Error registering 'xloBlock' (in FuncRegistry.cpp:372 during xloil::StaticWorksheetFunction::registerFunc)
[2023-08-24 16:58:36.945] [logger] [debug] [register.py:110] Interpreted arg '*Args' => 'Args:object (,vargs)'
[2023-08-24 16:58:36.945] [logger] [error] [FuncRegistry.cpp:385] Failed to register func xloBlock: Module 'xlOil_Utils' must define xlAutoFree12 to register Excel functions. Error registering 'xloBlock'
[2023-08-24 16:58:36.945] [logger] [debug] [register.py:431] Declared excel func: xloAttr(Object: object, Name: str, Args: object, Kwargs)
[2023-08-24 16:58:36.985] [logger] [debug] [Throw.cpp:17] Module 'xlOil_Utils' must define xlAutoFree12 to register Excel functions (in FuncRegistry.cpp:106 during xloil::FunctionRegistry::registerWithExcel)
[2023-08-24 16:58:36.985] [logger] [debug] [FuncRegistry.cpp:299] Deregistering xloConcat
</details>

This error occurs in **0.17.11** but not **0.17.10** from my testing using python 3.11.4 32-bit on 64bit Windows. 

I'm not a c++ guru but poked around. Here's what I've pieced together ( and apologies if this is super obvious ):

I think this is tied to commit 439e70829f5342a6c371d70ca9d6e9c76999cbd1 and that this [stack overflow post (x64 DLL export function names
) ](https://stackoverflow.com/questions/28062446/x64-dll-export-function-names
) discussing exported C++ name decoration highlights the issue.

 ```
      const auto autoFreeFunc = GetProcAddress(moduleHandle, "xlAutoFree12");
```
I believe under 32bit, the function name, ```"xlAutoFree12"```, needs to be updated. Since ```__stdcall``` is used, the name would be something like ```_xlAutoFree12@2``` rather than just prepending an underscore per ```__cdecl```. 

Unfortunately I don't know the best way to address this cleanly or exactly what the tradeoffs are between ```__cdecl``` or ```__stdcall```. [SO post on cdecl-or-stdcall-on-windows](https://stackoverflow.com/questions/6511096/cdecl-or-stdcall-on-windows) , [microsoft Calling Conventions docs](https://learn.microsoft.com/en-us/cpp/cpp/argument-passing-and-naming-conventions?view=msvc-170)

**Comment by cunnane** [comment]
Thank you, I'm glad you're enjoying the project!

Sorry, I missed this and didn't test the patch on the 32 bit build.  In this case the function needs to be undecorated/mangled to be found by Excel, so I forced the export name using a linker pragma.  This should be fixed in 0.17.13 which I released this week.

**Comment by andyvandy** [comment]
Much appreciated!! Still blown away by what you've built.
